<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="description" content="">
    <meta name="author" content="">
    <!--<link rel="icon" href="../../../../favicon.ico">-->

    <title>MusiteDeep Help</title>

    <!-- Bootstrap core CSS -->
    <!--<link href="../../../../dist/css/bootstrap.min.css" rel="stylesheet">-->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy" crossorigin="anonymous">
    <!-- Custom styles for this template -->
    
    <link href="help.css" rel="stylesheet">
    
   
    <script>
    function closeThis(){
    //   if (document.referrer != "") {
    //     window.opener = null;
    //   // window.open('', '_self');
    //   window.location.href="http://127.0.0.1:3000/";  
    //     history.back();
    //     window.close();
    // }
    // else{
    //   window.location.href="http://127.0.0.1:3000/";  
    // }
    
      
      window.location.href="http://www.musite.net";  
      //history.back();
      //window.close();
    }
    </script>
  </head>

  <body>
  
  
  
    <main role="main" class="container">


      
        <!-- Example row of columns -->
       
        <div>
          <h2 class="text-center">mRNALoc: a new deep-learning framework for multi-label mRNA subcellular localization prediction.</h2>
        </div>
        <div > 
            <button class="float-button" type="button" onclick="closeThis()">HOME</button> 
            <!-- float-button -->
        </div>
        
        <div>
        <p class="text-left" style="margin-top:50px;">Browser compatibility:</h3>
        <table class="browser"><tbody><td>OS</td><td>Version</td><td>chrome</td><td>Firefox</td><td>Microsoft Edge</td><td>Safari</td></tr>
                <tr><td>Linux</td><td>Ubuntu 16</td><td>75</td><td>70</td><td>n/a</td><td>n/a</td></tr>
                <tr><td>MacOS</td><td>HighSierra</td><td>77</td><td>70</td><td>n/a</td><td>12.0</td></tr>
                <tr><td>Windows</td><td>10</td><td>76</td><td>70</td><td>79</td><td>n/a</td></tr>
        </tbody>
        </table>
        <p class="text-jestify"></h3>
            Our server has a variety of models built for predicting post-translational modification sites (PTMs). 
            Users can simultaneously select multiple provided models for prediction using the multi-select drop down list.
            <img src="img/0.PNG" style="width: 100%;height: 5rem;">
        </p>
        
        
        <p class="text-left font-weight-bold">Input:</h3>
        
        
			<p class="text-justify">The only input needed is protein sequences in the <a href="https://zhanglab.ccmb.med.umich.edu/FASTA/">FASTA</a> format.  
           It accepts any alphabet characters, asterisks(*), hyphens(-), space, and line breaks. 
           All characters will be coded as a special character "X" excluding the 20 common amino acids represented by: 
          (A,C,D,E,F,G,H,I,K,L,M,N,P,Q,R,S,T,V,W,Y) for the prediction phase.             
          </p>
            
            <p class="text-justify">We provide two options for input：</p> 
          
            <p class="text-justify">
             1. Paste mode: users can paste up to 10 sequences or 5000 amino acids to the input panel. Upon clicking the <img style="height: 30px;width:150px" src="img/start.png"> button,
             the job will start immediately and return the result in real time. The job can be accessed later by checking the <img style="height: 30px;" src="img/userheader.png"> in the header.
            </p>
            
            
            <p class="text-justify">
            2   Upload mode: for a larger-scale prediction task (a FASTA file is 10MB at most), users can upload a file in the FASTA format by clicking the <span style="color: blue;">upload a FASTA file</span>. 
            In this mode, once the job is submitted, it will be placed on a queue for processing. The job can be accessed later by the provided URL or checking the the <img style="height: 30px;" src="img/userheader.png"> in the header. 
            In this mode, one user is allotted to process up to 5 jobs at the same time. 
            For large-scale jobs, we suggest to download and run the <a href="https://github.com/duolinwang/MusiteDeep_web">standalone tool</a>.
            </p>
            <p class="text-justify">For both modes, the jobs' inputs and results will be saved on the server for 72 hours (up to 100MB).</p>

            <p class="text-left font-weight-bold">Output:</h3>

            <p class="text-justify">
               After a job is finished, the result can be visualized for each input sequence one by one. 
            </p>

            <img src="img/1.png" style="width: 100%;height: 50vh;">
            <p></p>

            <p class="text-justify">
            The predicted PTMs are labeled using their abbreviations on the top of the corresponding positions. 
            Multiple labels are shown on top of one position if that position is predicted to have multiple PTMs. 
            The highlighted colors of the predicted sites correspond to their prediction confidence levels
            (the highest confidence is used for multiple labels).
            Upon hoovering the mouse on the predicited sites, the detailed information of the prediction will be shown.
            A user may adjust the prediction confidence threshold using the slider to obtain more or fewer predicted sites.
            <img src="img/2.png" style="width: 100%">
            The default threshold is set to 0.5, meaning sites with confidence scores higher than 0.5 are predicted and highlighted as PTM sites.
            </p>
            <p></p>
            
            <p class="text-justify">
              By clicking the <img src="img/3.png"> , the prediction results in the plain text will be saved.
            </p>
            <p>Here lists an example of the result file:</p>
            <div class="textholder">
               <div class="textcontent">
                <table>
                <tr><th>Position</th><th>Residue</th><th>PTMscores</th><th>Cutoff=0.5</th></tr>
                <tr><th>>sp|P97756|KKCC1_RAT</th></tr>
                <tr><th>3</th><th>R</th><th>Methylarginine:0.026</th><th>None</th></tr>
                <tr><th>4</th><th>S</th><th>Phosphoserine:0.847;O-linked_glycosylation:0.096</th><th>Phosphoserine:0.847</th></tr>
                <tr><th>5</th><th>P</th><th>Hydroxyproline:0.662</th><th>Hydroxyproline:0.662</th><tr>
                <tr><th>8</th><th>C</th><th>S-palmitoyl_cysteine:0.884</th><th>S-palmitoyl_cysteine:0.884</th></tr>
                <tr><th>9</th><th>C</th><th>S-palmitoyl_cysteine:0.799</th><th>S-palmitoyl_cysteine:0.799</th></tr>
                <tr><th>10</th><th>Q</th><th>Pyrrolidone_carboxylic_acid:0.111</th><th>None</th></tr>
                <tr><th>12</th><th>P</th><th>Hydroxyproline:0.06</th><th>None</th></tr>
                <tr><th>13</th><th>R</th><th>Methylarginine:0.018</th><th>None</th></tr>
                <tr><th>19</th><th>R</th><th>Methylarginine:0.016</th><th>None</th></tr>
                <tr><th>24</th><th>S</th><th>Phosphoserine:0.509;O-linked_glycosylation:0.047</th><th>Phosphoserine:0.509</th></tr>
                <tr><th>...</th><th>...</th><th>...</th><th>...</th></tr>
                <tr><th>>sp|A9QT41|NEMO_PIG</th></tr>
                <tr><th>2</th><th>S</th><th>Phosphoserine:0.103;O-linked_glycosylation:0.052</th><th>None</th></tr>
                <tr><th>3</th><th>R</th><th>Methylarginine:0.041</th><th>None</th></tr>
                <tr><th>4</th><th>T</th><th>Phosphothreonine:0.627;O-linked_glycosylation:0.041</th><th>Phosphothreonine:0.627</th></tr>
                <tr><th>5</th><th>P</th><th>Hydroxyproline:0.466</th><th>None</th></tr>
                <tr><th>8</th><th>S</th><th>Phosphoserine:0.412;O-linked_glycosylation:0.17</th><th>None</th></tr>
                <tr><th>9</th><th>Q</th><th>Pyrrolidone_carboxylic_acid:0.348</th><th>None</th></tr>
                <tr><th>10</th><th>P</th><th>Hydroxyproline:0.17</th><th>None</th></tr>
                <tr><th>11</th><th>C</th><th>S-palmitoyl_cysteine:0.642</th><th>S-palmitoyl_cysteine:0.642</th></tr>
                <tr><th>...</th><th>...</th><th>...</th><th>...</th></tr>
                </table>
            </div>
            </div>
            <p>Except for the first line of the file, which is the header, for each sequence, the first line is the sequence title (start with ">"), 
               the following lines are the prediction results of the potential PTM positions, 
               each line contains 4 columns and separated by the tab character ("\t"):</p>
            <p class="text-justify" style="padding:2em">
              1 Position: the position of the potential PTM sites.<br>
              2 Residue: the amino acid code of the residue at the position.<br>
              3 PTMscores: the potential PTMs and their predicted confidence scores; multiple PTMs are separated by semicolon.<br>
              4 Cutoffs=0.5: the predicted PTMs whose scores are higher than the present cutoff; the default cutoff is 0.5, which can be changed according to the user’s selection.<br>
            </p>
            
            <p class="text-left font-weight-bold">User job history:</h3>
            <p class="text-justify">
            All the users’ submitted jobs will be saved in the server for 72 hours (up to 100MB) 
            for the user to check anytime. By clicking <img src="img/userheader.png"> in the header, an example of the job history is shown below:
            <img src="img/userhistory.png" style="width:80%; margin: 10px 10px 10px 10px">
            <p>The jobs are listed in the chronological order of the job submission. 
               Users can veiw, download, and delete the jobs. 
               When viewing a job, the selected prediction models at the time of submission are shown as default selections for BLAST-BASED ANNOTATION. 
               Users can still change them to run Blast.
            </p>
            <p></p>
            <p class="text-left font-weight-bold">Dataset:</h3>

            <p class="text-justify">
              The preprocessed PTM data from the UniProt/Swiss-Prot database is provided to download.
              By clicking the  <img src="img/downloadptm.png" style="height:8vh"> in the header, it shows the following page. 
            </p>

            <p><img src="img/4.png" style="width: 80%;margin: 10px 10px 10px 10px"></p>
            <p></p>

            <p class="text-left font-weight-bold">Stand-alone tool:</h3>

            <p class="text-justify">
              We also provide a stand-alone tool, which can be accessed by clicking  
              <a href = "https://github.com/duolinwang/MusiteDeep_web"><img src="img/downloadtool.png" style="height:8vh"></a></p>
            
            <p class="text-left font-weight-bold">Other functions:</h3>
            <p class="text-justify">
              In the results panel, we provide several advanced functions:</p>
              <p>1 Blast-based annotation:</p>
              <p>This function will be activated by clicking <img src="img/blast.png">. It will do the Blast for the current sequence and presents its homologous sequences and their existing PTM annotations (from UniProt/Swiss-Prot database) in blue color according to the PTM type that the user select. </p>
              <p>The output is :</p>

            <img src="img/5.png" style="width: 100%">
            <p></p>

            <p class="text-justify">
              Users can refer to the sequence in UniProt/Swiss-Prot database by clicking the sequence title.
              Users can click the <img src="img/save_blast.PNG"> button to save the blast output to a file named "blastresult.txt". <br>
              Here shows an example of the result file:
            </p>
            <div class="textholder">
               <div class="textcontent" stype="font-size:12px">
               <p>>sp|P97756|KKCC1_RAT Calcium/calmodulin-dependent protein kinase kinase 1 OS=Rattus norvegicus GN=Camkk1 PE=1 SV=1</p>
               <p>MERSPAVCCQDPRAELVERVAAISVAHLEEAEEGPEPASNGVDPPPRARAASVIPGSASRPTPVRPSLSARKFSLQERPAGSCLEAQVGPYSTGPASHMSPRAWRRPTIESHHVAISDTEDCVQLNQYKLQSEIGKGAYGVVRLAYNEREDRHYAMKVLSKKKLLKQYGFPRRPPPRGSQAPQGGPAKQLLPLERVYQEIAILKKLDHVNVVKLIEVLDDPAEDNLYLVFDLLRKGPVMEVPCDKPFPEEQARLYLRDIILGLEYLHCQKIVHRDIKPSNLLLGDDGHVKIADFGVSNQFEGNDAQLSSTAGTPAFMAPEAISDTGQSFSGKALDVWATGVTLYCFVYGKCPFIDEYILALHRKIKNEAVVFPEEPEVSEELKDLILKMLDKNPETRIGVSDIKLHPWVTKHGEEPLPSEEEHCSVVEVTEEEVKNSVKLIPSWTTVILVKSMLRKRSFGNPFEPQARREERSMSAPGNLLLKEGCGEGGKSPELPGVQEDEAAS</p>
               <p>KKCC1_RAT (100)	67:Phosphoserine;74:Phosphoserine;78:Asymmetric dimethylarginine;100:Phosphoserine;108:Phosphothreonine;458:Phosphoserine;475:Phosphoserine;492:Phosphoserine;</p>
               <p>MERSPAVCCQDPRAELVERVAAISVAHLEEAEEGPEPASNGVDPPPRARAASVIPGSASRPTPVRPSLSARKFSLQERPAGSCLEAQVGPYSTGPASHMSPRAWRRPTIESHHVAISDTEDCVQLNQYKLQSEIGKGAYGVVRLAYNEREDRHYAMKVLSKKKLLKQYGFPRRPPPRGSQAPQGGPAKQLLPLERVYQEIAILKKLDHVNVVKLIEVLDDPAEDNLYLVFDLLRKGPVMEVPCDKPFPEEQARLYLRDIILGLEYLHCQKIVHRDIKPSNLLLGDDGHVKIADFGVSNQFEGNDAQLSSTAGTPAFMAPEAISDTGQSFSGKALDVWATGVTLYCFVYGKCPFIDEYILALHRKIKNEAVVFPEEPEVSEELKDLILKMLDKNPETRIGVSDIKLHPWVTKHGEEPLPSEEEHCSVVEVTEEEVKNSVKLIPSWTTVILVKSMLRKRSFGNPFEPQARREERSMSAPGNLLLKEGCGEGGKSPELPGVQEDEAAS</p>
               <p>KKCC1_MOUSE (97)	67:Phosphoserine;74:Phosphoserine;78:Asymmetric dimethylarginine;100:Phosphoserine;108:Phosphothreonine;458:Phosphoserine;475:Phosphoserine;492:Phosphoserine;</p>
               <p>MESGPAVCCQDPRAELVDRVAAINVAHLEEADEGPEPARNGVDPPPRARAASVIPGSASRPTPVRPSLSARKFSLQERPAGSCLGAQVGPYSTGPASHISPRSWRRPTIESHRVAISDTEDCVQLNQYKLQSEIGKGAYGVVRLAYNESEDRHYAMKVLSKKKLLKQYGFPRRPPPRGSQATQGGPAKQLLPLERVYQEIAILKKLDHVNVVKLIEVLDDPAEDNLYLVFDLLRKGPVMEVPCDKPFPEEQARLYLRDIILGLEYLHCQKIVHRDIKPSNLLLGDDGHVKIADFGVSNQFEGNDAQLSSTAGTPAFMAPEAISDSGQSFSGKALDVWATGVTLYCFVYGKCPFIDDYILTLHRKIKNEAVVFPEEPEVSEDLKDLILRMLDKNPETRIGVSDIKLHPWVTKHGEEPLPSEEEHCSVVEVTEEEVKNSVRLIPSWTTVILVKSMLRKRSFGNPFEPQARREERSMSAPGSLLMKEGCGEGCKSPELPGVQEDEAAS   </p>
               <p>KKCC1_HUMAN (96)	67:Phosphoserine;74:Phosphoserine;78:Asymmetric dimethylarginine;100:Phosphoserine;108:Phosphothreonine;458:Phosphoserine;475:Phosphoserine;492:Phosphoserine;                                                                                                                                                                                                                                                                                                                                            </p>
               <p>MEGGPAVCCQDPRAELVERVAAIDVTHLEEADGGPEPTRNGVDPPPRARAASVIPGSTSRLLPARPSLSARKLSLQERPAGSYLEAQAGPYATGPASHISPRAWRRPTIESHHVAISDAEDCVQLNQYKLQSEIGKGAYGVVRLAYNESEDRHYAMKVLSKKKLLKQYGFPRRPPPRGSQAAQGGPAKQLLPLERVYQEIAILKKLDHVNVVKLIEVLDDPAEDNLYLVFDLLRKGPVMEVPCDKPFSEEQARLYLRDVILGLEYLHCQKIVHRDIKPSNLLLGDDGHVKIADFGVSNQFEGNDAQLSSTAGTPAFMAPEAISDSGQSFSGKALDVWATGVTLYCFVYGKCPFIDDFILALHRKIKNEPVVFPEEPEISEELKDLILKMLDKNPETRIGVPDIKLHPWVTKNGEEPLPSEEEHCSVVEVTEEEVKNSVRLIPSWTTVILVKSMLRKRSFGNPFEPQARREERSMSAPGNLLVKEGFGEGGKSPELPGVQEDEAAS   </p>
               <p>KKCC2_RAT (79)	74:Phosphoserine;100:Phosphoserine;458:Phosphoserine;475:Phosphoserine;                                                                                                                                                                                                                                                                                                                                                                                                                                   </p>
               <p>-----------------------------------------------------LPLDASEPE-SRSLLSGGKMSLQERSQG------------GPASSSSPRMPRRPTVESHHVSITGLQDCVQLNQYTLKDEIGKGSYGVVKLAYNENDNTYYAMKVLSKKKLIRQAGFPRRPPPRGTRPAPGGCIQPRGPIEQVYQEIAILKKLDHPNVVKLVEVLDDPNEDHLYMVFELVNQGPVMEVPTLKPLSEDQARFYFQDLIKGIEYLHYQKIIHRDIKPSNLLVGEDGHIKIADFGVSNEFKGSDALLSNTVGTPAFMAPESLSETRKIFSGKALDVWAMGVTLYCFVFGQCPFMDERIMCLHSKIKSQALEFPDQPDIAEDLKDLITRMLDKNPESRIVVPEIKLHPWVTRHGAEPLPSEDENCTLVEVTEEEVENSVKHIPSLATVILVKTMIRKRSFGNPFE-GSRREERSLSAPGNLLTKK---------------------   </p>
               <p>KKCC2_MOUSE (78)	74:Phosphoserine;96:Phosphoserine;458:Phosphoserine;475:Phosphoserine;                                                                                                                                                                                                                                                                                                                                                                                                                                    </p>
               <p>------------------------------------------DQPPEADGQE-LPLEASDPESRSP-LSGRKMSLQE-------PSQGGPASSSNSLDMNGRCIRRPTVESHHVSITGLQDCVQLNQYTLKDEIGKGSYGVVKLAYNENDNTYYAMKVLSKKKLIRQAGFPRRPPPRGARPAPGGCIQPRGPIEQVYQEIAILKKLDHPNVVKLVEVLDDPNEDHLYMVFELVNQGPVMEVPTLKPLSEDQARFYFQDLIKGIEYLHYQKIIHRDIKPSNLLVGEDGHIKIADFGVSNEFKGSDALLSNTVGTPAFMAPESLSETRKIFSGKALDVWAMGVTLYCFVFGQCPFMDERIMCLHSKIKSQALEFPDQPDIAEDLKDLITRMLDKNPESRIVVPEIKLHPWVTRHGAEPLPSEDENCTLVEVTEEEVENSVKHIPSLATVILVKTMIRKRSFGNPFE-GSRREERSLSAPGNLLTKK---------------------   </p>
               <p>KKCC2_HUMAN (78)	74:Phosphoserine;83:Phosphoserine;92:Phosphoserine;96:Phosphoserine;100:Phosphoserine;458:Phosphoserine;475:Phosphoserine;                                                                                                                                                                                                                                                                                                                                                                                </p>
               <p>----------------------------------------------------------------RPHLSGRKLSLQERAAGGSLDMNICPYSPVSSPQSSPRLPRRPTVESHHVSITGMQDCVQLNQYTLKDEIGKGSYGVVKLAYNENDNTYYAMKVLSKKKLIRQAGFPRRPPPRGTRPAPGGCIQPRGPIEQVYQEIAILKKLDHPNVVKLVEVLDDPNEDHLYMVFELVNQGPVMEVPTLKPLSEDQARFYFQDLIKGIEYLHYQKIIHRDIKPSNLLVGEDGHIKIADFGVSNEFKGSDALLSNTVGTPAFMAPESLSETRKIFSGKALDVWAMGVTLYCFVFGQCPFMDERIMCLHSKIKSQALEFPDQPDIAEDLKDLITRMLDKNPESRIVVPEIKLHPWVTRHGAEPLPSEDENCTLVEVTEEEVENSVKHIPSLATVILVKTMIRKRSFGNPFE-GSRREERSLSAPGNLLTKK---------------------   </p>
               <p>KKCC_CAEEL (71)	                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          </p>
               <p>--------------------------------------------------------------------------------------EAGPHSSNNAATMSPNL-SRPTRYVKSVSQQRSESYIQLNQYRLMEEIGQGSYGIVKLAYNEEDKNLYALKVLDKMKLLKNFACFRQPPPRRNKE-NAAPSVLRNPLQLVQKEIAILKKLSHPNVVKLVEVLDDPNDNYLYMVFEFVEKGSILEIPTDKPLDEDTAWSYFRDTLCGLEYLHYQKIVHRDIKPSNLLLSDIGQVKIADFGVSCEFEGIDAFLSGTAGTPAFMAPEALTEGANHFSGRAQDIWSLGITLYAFVIGTVPFVDNYIIALHKKIKNDPIVFPEAPILSEALQDIILGMLKKDPGHRLMLHEVKVHTWVTRDGTVPMSSEQENCHLVTVTEEEIENCVRVIPRLDTLILVKAMGHRKRFGNPF------------------------------------------   </p>
               </div>
            </div>
            <div>
            <p class="text-justify">            
            The first two lines represent the ID and sequence of the query sequence, and the following lines are the aligned sequences by Blast with the PTM annotations in the Uniprot/Swiss-Prot database. 
              Each alignment result consists of two lines: the first line contains the Uniprot sequence access ID, 
              the Blast identity in the parenthesis, 
              the selected PTM types for annotations and the corresponding positions 
              (according to the position of the query sequence) that have the annotations;
              the second line contains the aligned amino acids, with hyphens "-" representing the unaligned ones.
              
              
             <p>2 View predicted PTM sites in 3D structure:</p>
             <p>This function provides the visualization of the predicted PTM sites in the context of protein 3D structure. Once click the 
             <img src="img/show3d.PNG"> button, the list of homologous proteins that have known 3D structures will be shown:

            </p>

            <img class="jpg6" src="img/6.png">
            <P></P>
            
            <p>
            <p class="text-justify">
              Users can choose one protein and by clicking the <img src="img/showedbutton.png"> button the 3D structure with the information of the predicted PTM sites will be shown in a new page.
            </p>
            <p>Here is an example:</p>
            <img class="jpg7" src = "img/7.png">
            <p>The highlighted labels represent the information of the predicted sites, which contains the position on the query sequence, 
            the amino acid from the query sequence to the mapped PDB sequence, and the predicted PTM types (in abbreviation) for this site. Multiple PTMs are separated by comma ",". 
            The default representation of the PTM sites and structrues will be shown at the first time and users can make slightly changes by tuning the parameters. Users are suggested to add the surface by checking 
            the "Add surface" box. 
            </p>
            </p>
            <p></p>
            <p></p>

            
              

            
  
        
        
        

      </div> <!-- /container -->

    </main>

    
  </body>
</html>
